ha_healthchecker/README.md

```diff
@@ -0,0 +1,146 @@
+    
+    `Master` is the cluster that OpenLab runs on.
+    
+    `Slave` is the cluster for backup, once master is down, `ha_healthchecker` will switch OpenLab from Master to Slave to ensure OpenLab works. 
+
+    `Zookeeper` is the node that only runs zookeeper service which is required by zookeeper cluster deployment mode.
+
+2. Zuul/Nodepool/Zookeeper
+
+    There are three node types in OpenLab HA development.
+
+    `Zuul` is the node that runs zuul related services.
+
+    `Nodepool` is the node that runs nodepool related services.
+
+    `Zookeeper `is the node that only runs zookeeper service. It's role is `zookeeper` as well.
+
+3. Necessary/Unnecessary Service.
+  
+    There are many services run on OpenLab nodes.
+    
+    A necessary service is a service that must be always ran. Once it's down, OpenLab will be down and won't work any more. 
+
+    An unnecessary service can be down for a period of time. (Default is 2 days). After that time, if it's still down, OpenLab will be treat as down.
+    
+    The service mapping is shown as below.
+    ```
+    service_mapping = {
+        'master': {
+            'zuul': {
+                'necessary': ['zuul-scheduler', 'zuul-executor', 'zuul-web',
+                              'gearman-job-server', 'mysql', 'apache2'],
+                'unnecessary': ['zuul-merger', 'zuul-fingergw', 'zuul-timer-tasks']
+            },
+            'nodepool': {
+                'necessary': ['nodepool-launcher'],
+                'unnecessary': ['nodepool-timer-tasks', 'nodepool-builder',
+                                'zookeeper']
+            }
+        },
+        'slave': {
+            'zuul': {
+                'necessary': [],
+                'unnecessary': ['mysql', 'rsyncd']
+            },
+            'nodepool': {
+                'necessary': [],
+                'unnecessary': ['zookeeper', 'rsyncd']
+            }
+        },
+        'zookeeper': {
+            'zookeeper': {
+                'necessary': [],
+                'unnecessary': ['zookeeper']
+            }
+        },
+    }
+    ```
+
+## How it works
+
+`ha_healthchecker` now includes **Refresh**, **Fix** and **Switch** functions. More will be added in the future if needed.
+
+* **Refresh**
+
+    `ha_healthchecker` checks openlab nodes and the services which run on them every 2 minutes. If everything is OK, the nodes/services' heartbeat will be refreshed. Otherwise
+    the service will be marked as `restarting` or `down`.
+
+* **Fix**
+
+    Once any service hits error(that's said the service status in systemd is not **running**), `ha_healthchecker` will do the fix step. Here are some cases:
+
+    1. If the service is a necessary service.
+  
+        a. If the service is marked as `restaring`, it'll be restarted max 3 times (by default).
+
+        b. If the service is marked as `down`, `ha_healthchecker` will send a github issue to `openlab` repo and try to start switching Master and Slave.
+
```

Comment:
Just a question: is there any other status except restarting and down?

Reply:
Good question. I'll add a status matrix section